numba/tests/test_parfors.py

```diff
@@ -964,6 +972,283 @@ def test_impl(n):
+        def test_impl(a):
+            (m,n) = a.shape
+            b = a.transpose()
+            b[1:m,1:n] = a[1:m,1:n]
+            return b
+
+        self.check(test_impl, np.arange(9).reshape((3,3)))
+
+    @skip_unsupported
+    def test_parfor_slice9(self):
+        def test_impl(a):
+            (m,n) = a.shape
+            b = a.transpose()
+            b[1:n,1:m] = a[:,1:m]
+            return b
+
+        self.check(test_impl, np.arange(12).reshape((3,4)))
+
+    @skip_unsupported
+    def test_parfor_slice10(self):
+        def test_impl(a):
+            (m,n) = a.shape
+            b = a.transpose()
+            b[2,1:m] = a[2,1:m]
+            return b
+
+        self.check(test_impl, np.arange(9).reshape((3,3)))
+
+    @skip_unsupported
+    def test_parfor_slice11(self):
+        def test_impl(a):
+            (m,n,l) = a.shape
+            b = a.copy()
+            b[:,1,1:l] = a[:,2,1:l]
+            return b
+
+        self.check(test_impl, np.arange(27).reshape((3,3,3)))
+
+    @skip_unsupported
+    def test_parfor_slice12(self):
+        def test_impl(a):
+            (m,n) = a.shape
+            b = a.copy()
+            b[1,1:-1] = a[0,:-2]
+            return b
+
+        self.check(test_impl, np.arange(12).reshape((3,4)))
+
+    @skip_unsupported
+    def test_parfor_slice13(self):
+        def test_impl(a):
+            (m,n) = a.shape
+            b = a.copy()
+            c = -1
+            b[1,1:c] = a[0,-n:c-1]
+            return b
+
+        self.check(test_impl, np.arange(12).reshape((3,4)))
+
+    @skip_unsupported
+    def test_parfor_slice14(self):
+        def test_impl(a):
+            (m,n) = a.shape
+            b = a.copy()
+            c = -1
+            b[1,:-1] = a[0,-3:4]
+            return b
+
+        self.check(test_impl, np.arange(12).reshape((3,4)))
+
+    @skip_unsupported
+    def test_parfor_slice15(self):
+        def test_impl(a):
+            (m,n) = a.shape
+            b = a.copy()
+            c = -1
+            b[1,-(n-1):] = a[0,-3:4]
+            return b
+
+        self.check(test_impl, np.arange(12).reshape((3,4)))
+
+    @skip_unsupported
+    def test_parfor_slice16(self):
+        def test_impl(a, b, n):
+            assert(a.shape == b.shape)
+            a[1:n] = 10
+            b[0:(n-1)] = 10
+            return a * b
+
+        self.check(test_impl, np.ones(10), np.zeros(10), 2)
+        args = (numba.float64[:], numba.float64[:], numba.int64)
+        self.assertEqual(countParfors(test_impl, args), 2)
+
+
+class TestParforsOptions(TestParforsBase):
+
+    def check(self, pyfunc, *args, **kwargs):
+        cfunc, cpfunc = self.compile_all(pyfunc, *args)
+        self.check_prange_vs_others(pyfunc, cfunc, cpfunc, *args, **kwargs)
+
+    @skip_unsupported
+    def test_parfor_options(self):
+        def test_impl(a):
+            n = a.shape[0]
+            b = np.ones(n)
+            c = np.array([ i for i in range(n) ])
+            # a[:n] = a + b * c
```

Comment:
delete?

Reply:
There is actually a bug with fusion if this is turned on. I'm looking into fixing it.

Reply:
False alarm. It turns out to be the test was mutating its input, which causes repeated runs to have different outputs. Now it is fixed.